web/ui/src/Pools/PoolsController.js

```diff
@@ -6,33 +6,110 @@
 (function() {
     'use strict';
 
-    controlplane.controller("PoolsController", ["$scope", "$routeParams",
-    "resourcesFactory", "authService", "$modalService", "$translate",
-    "$notification", "areUIReady", "$interval", "servicedConfig", "log",
-    "POOL_PERMISSIONS", "miscUtils", "Pool",
-    function($scope, $routeParams, resourcesFactory, authService, $modalService,
-    $translate, $notification, areUIReady, $interval, servicedConfig, log,
-    POOL_PERMISSIONS, utils, Pool){
-
-        // Ensure logged in
-        authService.checkLogin($scope);
-
-        // allow templates to get the list
-        // of permissions
-        $scope.permissions = POOL_PERMISSIONS;
-
-        $scope.clickPool = function(id) {
-            resourcesFactory.routeToPool(id);
-        };
-
-        // Function to remove a pool
-        $scope.clickRemovePool = function(poolID) {
-            if ($scope.isDefaultPool(poolID)) {
-              return;
+    class PoolsController {
+
+        constructor ($scope, resourcesFactory, authService, $modalService, 
+                     $translate, $notification, areUIReady, $interval, servicedConfig, 
+                     log, POOL_PERMISSIONS, miscUtils, Pool, $q)
+        {
+            authService.checkLogin(this);
+
+            this.resourcesFactory = resourcesFactory;
+            this.$modalService = $modalService;
+            this.$translate = $translate;
+            this.$notification = $notification;
+            this.areUIReady = areUIReady;
+            this.$interval = $interval;
+            this.utils = miscUtils;
+            this.$q = $q;
+            this.permissions = POOL_PERMISSIONS;
+            this.name = "pools";
```

Comment:
Please use the pattern described in a comment on HostsController.js. to clean up this dependency stuff.